gr-utils/python/blocktool/core/parseheader.py

```diff
@@ -0,0 +1,278 @@
+                                                    header_file=self.info['target_file'])
+            criteria = declarations.calldef_matcher(name='make')
+            _make_fun = declarations.matcher.get_single(criteria, main_class)
+            _make_fun = str(_make_fun).split(
+                'make')[-1].split(')')[0].split('(')[1].lstrip().rstrip().split(',')
+            if make_func:
+                for arg in make_func[0].arguments:
+                    for _arg in _make_fun:
+                        if str(arg.name) in _arg:
+                            make_arguments = {
+                                "name": str(arg.name),
+                                "dtype": str(arg.decl_type),
+                                "default": ""
+                            }
+                            if re.findall(r'[-+]?\d*\.\d+|\d+', _arg):
+                                make_arguments['default'] = re.findall(
+                                    r'[-+]?\d*\.\d+|\d+', _arg)[0]
+                            elif re.findall(r'\"(.+?)\"', _arg):
+                                make_arguments['default'] = re.findall(
+                                    r'\"(.+?)\"', _arg)[0]
+                            elif "true" in _arg:
+                                make_arguments['default'] = "True"
+                            elif "false" in _arg:
+                                make_arguments['default'] = "False"
+                    self.parsed_data['make']['arguments'].append(
+                        make_arguments.copy())
+        except BlockToolException as exception:
+            raise BlockToolException(
+                'A block API always has a factory signature!\n{}'.format(exception))
+
+        # setters
+        try:
+            self.parsed_data['methods'] = []
+            query_methods = declarations.access_type_matcher_t('public')
+            setters = main_class.member_functions(function=query_methods,
+                                                  allow_empty=True,
+                                                  header_file=self.info['target_file'])
+            getter_arguments = []
+            if setters:
+                for setter in setters:
+                    if str(setter.name).startswith('set_') and setter.arguments:
+                        setter_args = {
+                            "name": str(setter.name),
+                            "arguments_type": []
+                        }
+                        for argument in setter.arguments:
+                            args = {
+                                "name": str(argument.name),
+                                "dtype": str(argument.decl_type)
+                            }
+                            getter_arguments.append(args['name'])
+                            setter_args['arguments_type'].append(args.copy())
+                        self.parsed_data['methods'].append(setter_args.copy())
+        except:
```

Comment:
Wouldn't it be a bit obscure for the API user if the exception doesn't `raises` or produce a type of input to work with? it seems that when the getters and setters fail we end up, unknowingly, with empty parsed data

Reply:
This was one of the doubts I had and wanted to clear with you. Sure, an exception can be raised.